### DISCUSSION TITLE
DMN 1.4 - scales

### DISCUSSION THREAD

**Discussion by StrayAlien** [dmn-tck/tck#607] [General]
Hey all, with the DMN 1.4 PRs here now, there are quite a number that deal with numeric scales.

Admittedly, my understanding of how we're supposed to handle negative scales is pretty patchy.  Hence leaving off tests until I / we all have an understanding.  TBH, it is not is much "what is a negative scale", that is okay.  More like, "how to transition between scales when a negitve is involved".  Debugging through the Java BigDecimal code is is not straight forward at all.

So, opening up a discussion here so we can all understand better.

First helpful comment is by @opatrascoiu here: https://github.com/dmn-tck/tck/pull/592#issuecomment-1607414086. :-)

**Comment by opatrascoiu** [comment]
According to DMN spec, Section 10.3.2.3.1 number

```
FEEL Numbers are based on IEEE 754-2008 Decimal128 format, with 34 decimal digits of precision and rounding
toward the nearest neighbor with ties favoring the even neighbor. Numbers are a restriction of the XML Schema
type precisionDecimal, and are equivalent to Java BigDecimal with MathContext DECIMAL 128.

```

Looks obvious to me the tests should match the setScale() method for BigDecimal with MathContext DECIMAL 128.

I think the same behavior as java BigDecimal can be achieved in Python by using decimal.Decimal. I have identical negative scale tests running both on Java and Python.

**Comment by opatrascoiu** [comment]
According to this https://docs.oracle.com/javase/8/docs/api/java/math/BigDecimal.html the scale is a 32 bit signed integer.

Hence, the tests that cover overflow / underflow of scale should use the -2,147,483,648 and 2,147,483,647.

**Comment by opatrascoiu** [comment]
Following our discussion today, I created two DMN issues to cover the scale issues:
https://issues.omg.org/browse/DMN16-81
https://issues.omg.org/browse/DMN16-82
